python/ml4ir/applications/ranking/tests/test_losses.py

```diff
@@ -62,3 +62,12 @@ def test_rank_one_listnet(self):
         loss = self.run_default_pipeline(loss_key="rank_one_listnet")
 
         assert np.isclose(loss, 2.00879, rtol=0.05)
+
+    def test_softmax_cross_entropy(self):
+        """
+        Test model training and evaluate Rank One ListNet loss
+        """
+
+        loss = self.run_default_pipeline(loss_key="softmax_cross_entropy")
+
+        assert np.isclose(loss, 1.19955, rtol=0.05)
```

Comment:
These tests, where we run the pipeline and get `np.isclose` to something, have value, in the sense that the pipeline runs. They boost coverage too! 
This is not the whole story: ideally we want to eval the actual code in `SoftmaxCrossEntropy`  and `get_loss_fn` where the tensor calculations happen. Would you consider adding such a "toy" test?

Reply:
I agree. I am going to change all of them to just unit tests actually. It will help with memory leak as well. I think the current build is failing because of memory overflow.

Reply:
Did you have memory issues when training actual models (not toy in tests)? I have been training for several days but without issues for instance (which is good). Perhaps I had more memory.. But I remember having issues with tests (locally)

Reply:
Not on my local box. I think it only breaks on the autobuilds because of the docker container hard limitation. Our local boxes have a lot of swap size for handling this (is my guess). 

The main issue is that in tf2, keras models are not releasing memory. So for a test like this, we create the pipeline many times (so we are creating and deleting keras models) and the model is not getting "garbage collected" fully. :|